### ISSUE TITLE
Null safety warnings with Flutter 3 / Dart 2.17 

### ISSUE THREAD

**Issue by Zazo032** [ueman/feedback#200]
### Version

2.4.1

### Library

feedback

### Flutter channel

stable

### Flutter version

3.0.0

### Platform

Android

## Logs
<details>

```sh
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/back_button_interceptor.dart:44:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.addObserver(instance);
                   ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/back_button_interceptor.dart:49:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.removeObserver(instance);
                   ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/l18n/localization.dart:44:41: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
  final Locale? locale = WidgetsBinding.instance?.window.locale;
                                        ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/media_query_from_window.dart:25:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.addObserver(this);
                   ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/media_query_from_window.dart:70:54: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
      data: MediaQueryData.fromWindow(WidgetsBinding.instance!.window),
                                                     ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/media_query_from_window.dart:77:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.removeObserver(this);
                   ^

/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/media_query_from_window.dart:25:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.addObserver(this);
                   ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/media_query_from_window.dart:70:54: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
      data: MediaQueryData.fromWindow(WidgetsBinding.instance!.window),
                                                     ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/media_query_from_window.dart:77:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.removeObserver(this);
                   ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/l18n/localization.dart:44:41: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
  final Locale? locale = WidgetsBinding.instance?.window.locale;
                                        ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/back_button_interceptor.dart:44:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.addObserver(instance);
                   ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/back_button_interceptor.dart:49:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.removeObserver(instance);
                   ^

/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/media_query_from_window.dart:25:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.addObserver(this);
                   ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/media_query_from_window.dart:70:54: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
      data: MediaQueryData.fromWindow(WidgetsBinding.instance!.window),
                                                     ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/media_query_from_window.dart:77:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.removeObserver(this);
                   ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/l18n/localization.dart:44:41: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
  final Locale? locale = WidgetsBinding.instance?.window.locale;
                                        ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/back_button_interceptor.dart:44:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.addObserver(instance);
                   ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/feedback-2.4.1/lib/src/utilities/back_button_interceptor.dart:49:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.removeObserver(instance);
```
</details>

### Steps to reproduce

- Upgrade to Flutter 3
- Set minimum `environment' to 2.17:
```yaml
environment:
  sdk: ">=2.17.0 <3.0.0"
```
- Run the example

### Output of `flutter doctor -v`

```shell
[✓] Flutter (Channel stable, 3.0.0, on Microsoft Windows [Versi¢n 10.0.22000.675], locale es-ES)
    • Flutter version 3.0.0 at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (2 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc3)
    • Android SDK at C:\Users\crist\AppData\Local\Android\Sdk
    • Platform android-Tiramisu, build-tools 33.0.0-rc3
    • ANDROID_HOME = C:\Users\crist\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[✓] VS Code (version 1.66.2)
    • VS Code at C:\Users\crist\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.39.20220405

[✓] Connected device (4 available)
    • Pixel XL (mobile)            • HT69W0202715  • android-arm64  • Android 10 (API 29)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 12 (API 32) (emulator)
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 101.0.4951.54
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 101.0.1210.39

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by ueman** [comment]
I think I already fixed some of them a while ago, but didn't come around to release it. Thanks for the report, that was quick :D

**Comment by renanyoy** [comment]
I'm waiting the fix so ;)

**Comment by ueman** [comment]
Just released version 2.5 which fixes this.
